src/app/clusters/ethernet-network-diagnostics-server/ethernet-diagnostics-cluster.cpp

```diff
@@ -0,0 +1,228 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app/clusters/ethernet-network-diagnostics-server/ethernet-diagnostics-cluster.h>
+
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+
+#include <clusters/EthernetNetworkDiagnostics/Attributes.h>
+#include <clusters/EthernetNetworkDiagnostics/Commands.h>
+#include <clusters/EthernetNetworkDiagnostics/Ids.h>
+#include <clusters/EthernetNetworkDiagnostics/Metadata.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+using namespace chip::app::Clusters::EthernetNetworkDiagnostics;
+
+namespace {
+
+CHIP_ERROR ReadPHYRate(DeviceLayer::DiagnosticDataProvider & provider, AttributeValueEncoder & encoder)
+{
+    EthernetNetworkDiagnostics::Attributes::PHYRate::TypeInfo::Type pHYRate;
+    auto value = EthernetNetworkDiagnostics::PHYRateEnum::kRate10M;
+
+    if (provider.GetEthPHYRate(value) == CHIP_NO_ERROR)
+    {
+        pHYRate.SetNonNull(value);
+    }
+
+    return encoder.Encode(pHYRate);
+}
+
+CHIP_ERROR ReadFullDuplex(DeviceLayer::DiagnosticDataProvider & provider, AttributeValueEncoder & encoder)
+{
+    EthernetNetworkDiagnostics::Attributes::FullDuplex::TypeInfo::Type fullDuplex;
+    bool value = false;
+
+    if (provider.GetEthFullDuplex(value) == CHIP_NO_ERROR)
+    {
+        fullDuplex.SetNonNull(value);
+    }
+    else
+    {
+        ChipLogProgress(Zcl, "The Ethernet interface is not currently configured or operational");
```

Comment:
nit: do we need these logs? We do not log on having a value, so why is there a log on not having one. When we query, the client sees the content of the message and knows there is no value.

Since we try to preserve flash as much as possible, I would maybe reconsider having these log lines. Ok either way though (assume logs were pre-existing).